### PR TITLE
Rename/Update PowerShell ETW manifest to remove the Windows PowerShell dependency.

### PR DESCRIPTION
**Pull request by dantraMSFT**
This is the first part of the fix for https://github.com/PowerShell/PowerShell/issues/4939 and include the following changes:
1: Rename the manifest file (PowerShell.Core.Instrumentation.man)
2: ~Change the ETW provider GUID and name and move to outside of the Microsoft/Windows tree in the event log. The provider name is now simply PowerShellCore.~ [edited by @daxian-dbw: provider GUID and provider names are not changed in this PR] 
3: Create a registration script for registering. (RegisterManifest.ps1)
4: Produce a binary resource-only DLL to contain the ETW manifest and associated string resources. (PowerShell.Core.Instrumentation.dll)

The second part will involve creating a nuget package that will be pulled in at build time, similar to psl-native. for Linux  Additionally, tests to verify the provider registration and event raising will be included as part of the nuget package work.

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -257,6 +257,36 @@ cmd.exe /C cd /d "$location" "&" "$($vcPath)\vcvarsall.bat" "$Arch" "&" cmake "$
             log "  Copying $srcPath to $dstPath"
             Copy-Item $srcPath $dstPath
         }
+
+        # Place the remoting configuration script in the same directory
+        # as the binary so it will get published.
+        Copy-Item .\Install-PowerShellRemoting.ps1 $dstPath
```
Comment (daxian-dbw) [inline build.psm1]: Bin placing `Install-PowerShellRemoting.ps1` has been removed in #5330

File: build.psm1 [hunk 2]
```diff
@@ -257,6 +257,36 @@ cmd.exe /C cd /d "$location" "&" "$($vcPath)\vcvarsall.bat" "$Arch" "&" cmake "$
...
+        Set-Location -Path $location
+
+        $command = @"
+cmd.exe /C cd /d "$location" "&" "$($vcPath)\vcvarsall.bat" "$Arch" "&" cmake "$overrideFlags" -DBUILD_ONECORE=ON -DBUILD_TARGET_ARCH=$Arch -G "$cmakeGenerator" . "&" msbuild ALL_BUILD.vcxproj "/p:Configuration=$Configuration"
+"@
+        log "  Executing Build Command for PowerShell.Core.Instrumentation: $command"
+        Start-NativeExecution { Invoke-Expression -Command:$command }
+
+        # Copy the binaries and manifest to the packaging directory
+        $dstPath = "$PSScriptRoot\src\powershell-win-core"
```
Comment (daxian-dbw) [inline build.psm1]: This line seems unnecessary.

**Comment by daxian-dbw** [comment]
The manifest has some windows powershell specific resources. Should this PR include the cleanup work of those resources?

**Comment by SteveL-MSFT** [comment]
@daxian-dbw I think we can do the cleanup of the manifest in a separate PR

**Comment by dantraMSFT** [comment]
@daxian-dbw I did do an initial scrub by removing all events that are not referenced in code (i.e, scanning for all PSEvents.* code references).  Going further would require removing Windows-specific code which I thought would only muddle the PR.

**Comment by dantraMSFT** [comment]
@daxian-dbw: I made you the assignee; let me know if it should be someone else. I need to create a nuget package once this is merged.
Thanks,

File: src/PowerShell.Core.Instrumentation/RegisterManifest.ps1 [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    if (-not (Test-Path -Path $file))
+    {
+        throw "Could not find $($file.Name) at $Path"
+    }
+}
+
+[string] $command = $null
+if ($Unregister)
+{
+    $command = [string]::Format("wevtutil um {0}", $manifest.FullName)
```
Comment (daxian-dbw) [inline src/PowerShell.Core.Instrumentation/RegisterManifest.ps1]: Please use `"wevtutil um {0}" -f $manifest.FullName` for the formatting.

File: src/PowerShell.Core.Instrumentation/RegisterManifest.ps1 [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+}
+else
+{
+    $command = [string]::Format("wevtutil.exe im {0} /rf:{1} /mf:{1}", $manifest.FullName, $binary.FullName)
```
Comment (daxian-dbw) [inline src/PowerShell.Core.Instrumentation/RegisterManifest.ps1]: Same here.

**Comment by daxian-dbw** [comment]
@TravisEz13 @adityapatwardhan Can you please also review this PR?

File: build.psm1 (hunk 2 above)
Comment (dantraMSFT) [reply build.psm1]: No, but I don't want to assume the previous build step sets it as expected.

File: build.psm1 (hunk 1 above)
Comment (dantraMSFT) [reply build.psm1]: Fixed

File: src/PowerShell.Core.Instrumentation/RegisterManifest.ps1 (hunk 3 above)
Comment (dantraMSFT) [reply src/PowerShell.Core.Instrumentation/RegisterManifest.ps1]: Fixed

File: src/PowerShell.Core.Instrumentation/RegisterManifest.ps1 (hunk 4 above)
Comment (dantraMSFT) [reply src/PowerShell.Core.Instrumentation/RegisterManifest.ps1]: Fixed

File: build.psm1 (hunk 2 above)
Comment (daxian-dbw) [reply build.psm1]: Fair enough :)

File: build.psm1 [hunk 5]
```diff
@@ -257,6 +257,32 @@ cmd.exe /C cd /d "$location" "&" "$($vcPath)\vcvarsall.bat" "$Arch" "&" cmake "$
...
+        $FilesToCopy = @(
+            [IO.Path]::Combine($location, $Configuration, 'PowerShell.Core.Instrumentation.dll'),
+            [IO.Path]::Combine($location, 'PowerShell.Core.Instrumentation.man')
+            [IO.Path]::Combine($location, 'RegisterManifest.ps1')
```
Comment (daxian-dbw) [inline build.psm1]: Do not bin-place the `.ps1` and `.man` file. Instead, we can copy them to `publish` folder directly by `powershell-win-core.csproj` file. See how we bin-place `install-powershellremoting` [here](https://github.com/PowerShell/PowerShell/blob/master/src/powershell-win-core/powershell-win-core.csproj#L22).

In this way, we don't need to add a `.gitignore` file in the `powershell-win-core` folder to ignore the `.man` file there.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: build.psm1 (hunk 5 above)
Comment (dantraMSFT) [reply build.psm1]: @daxian-dbw: Does this also mean they shouldn't be included in the nuget package?

Comment (dantraMSFT) [reply build.psm1]: Fixed

**Comment by dantraMSFT** [comment]
@TravisEz13 @adityapatwardhan Do you have any feedback?

**Review by daxian-dbw — APPROVED** [review]

File: src/PowerShell.Core.Instrumentation/version.rc [hunk 6]
```diff
@@ -0,0 +1,15 @@
+//
+//    Copyright (C) Microsoft.  All rights reserved.
+//
+#include <windows.h>
+#include <ntverp.h>
+
+#define VER_FILETYPE    		VFT_DLL
+#define VER_FILESUBTYPE 		VFT2_UNKNOWN
+#define VER_FILEDESCRIPTION_STR     	"DSC"
```
Comment (adityapatwardhan) [inline src/PowerShell.Core.Instrumentation/version.rc]: Should this be PowerShellCore?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: src/PowerShell.Core.Instrumentation/version.rc (hunk 6 above)
Comment (dantraMSFT) [reply src/PowerShell.Core.Instrumentation/version.rc]: Fixed

Comment (daxian-dbw) [reply src/PowerShell.Core.Instrumentation/version.rc]: @dantraMSFT please push your fix so that @adityapatwardhan can sign off.

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Comment by dantraMSFT** [comment]
@daxian-dbw This should be good to go.

**Comment by daxian-dbw** [comment]
Test failure in AppVeyor is a known issue, but Travis CI hasn't finished yet.

**Pull request merged**